packages/wrangler/src/r2/sippy.ts

```diff
@@ -0,0 +1,224 @@
+import { readConfig } from "../config";
+import { prompt } from "../dialogs";
+import { UserError } from "../errors";
+import { logger } from "../logger";
+import { APIError, readFileSync } from "../parse";
+import { requireAuth } from "../user";
+import { deleteR2Sippy, getR2Sippy, putR2Sippy } from "./helpers";
+import type {
+	CommonYargsArgv,
+	StrictYargsOptionsToInterface,
+} from "../yargs-types";
+import type { SippyPutConfig } from "./helpers";
+
+export function EnableOptions(yargs: CommonYargsArgv) {
+	return yargs
+		.positional("name", {
+			describe: "The name of the bucket",
+			type: "string",
+			demandOption: true,
+		})
+		.option("jurisdiction", {
+			describe: "The jurisdiction where the bucket exists",
+			alias: "J",
+			requiresArg: true,
+			type: "string",
+		})
+		.option("provider", {
+			choices: ["AWS", "GCS"],
+		})
+		.option("bucket", {
+			description: "The name of the upstream bucket",
+			string: true,
+		})
+		.option("region", {
+			description: "(AWS provider only) The region of the upstream bucket",
+			string: true,
+		})
+		.option("key-id", {
+			description:
+				"(AWS provider only) The secret access key id for the upstream bucket",
+			string: true,
+		})
+		.option("secret-access-key", {
+			description:
+				"(AWS provider only) The secret access key for the upstream bucket",
+			string: true,
+		})
+		.option("service-account-key-file", {
+			description:
+				"(GCS provider only) The path to your Google Cloud service account key JSON file",
+			string: true,
+		})
+		.option("client-email", {
+			description:
+				"(GCS provider only) The client email for your Google Cloud service account key",
+			string: true,
+		})
+		.option("private-key", {
+			description:
+				"(GCS provider only) The private key for your Google Cloud service account key",
+			string: true,
+		})
+		.option("r2-key-id", {
+			description: "The secret access key id for this R2 bucket",
+			string: true,
+		})
+		.option("r2-secret-access-key", {
+			description: "The secret access key for this R2 bucket",
+			string: true,
+		});
+}
+
+export async function EnableHandler(
+	args: StrictYargsOptionsToInterface<typeof EnableOptions>
+) {
+	const isInteractive = process.stdin.isTTY;
+	const config = readConfig(args.config, args);
+	const accountId = await requireAuth(config);
+
+	if (isInteractive) {
+		args.provider ??= await prompt(
```

Comment:
It looks like there's no handling to ensure that the user fills in a valid value here or later on - could result in a weird error